### PR TITLE
cmd: add --all flag to kibana resync command

### PR DESCRIPTION
**Pull request by karencfv**
## Description
Implements a new  `--all` flag for kibana resync:

```console
$ ecctl deployment kibana resync --all
```

## Related Issues
Closes: https://github.com/elastic/cloud-cli/issues/1017

## How Has This Been Tested?
Manual and Unit tests

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improves code quality but has no user-facing effect)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Readiness Checklist

- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: cmd/deployment/kibana/resync.go [hunk 1]
```diff
@@ -28,14 +28,31 @@ import (
 )
 
 var resyncKibanaCmd = &cobra.Command{
-	Use:     "resync <cluster id>",
+	Use:     "resync [<deployment id> | --all]",
 	Short:   "Resynchronizes the search index and cache for the selected Kibana instance",
```
Comment (ppapapetrou76) [inline cmd/deployment/kibana/resync.go]:
```suggestion
	Short:   "Resynchronizes the search index and cache for the selected Kibana instance or all instances",
```

File: cmd/deployment/kibana/resync.go [hunk 2]
```diff
@@ -28,14 +28,31 @@ import (
...
+				API: ecctl.Get().API,
+			})
+			if err != nil {
+				return err
+			}
+
+			return ecctl.Get().Formatter.Format("", res)
+		}
+
 		fmt.Printf("Resynchronizing Kibana instance: %s\n", args[0])
```
Comment (ppapapetrou76) [inline cmd/deployment/kibana/resync.go]: Do we want to add a similar message when syncing all kibanas for consistency?

File: pkg/deployment/kibana/resync_test.go [hunk 3]
```diff
@@ -102,3 +103,67 @@ func TestResync(t *testing.T) {
...
+				}),
+			}},
+			wantErr: &url.Error{
+				Op:  "Post",
+				URL: "https://mock-host/mock-path/clusters/kibana/_resync?skip_matching_version=true",
+				Err: errors.New("error with API"),
+			},
+		},
+		{
+			name: "Succeeds to resynchronize Kibana instance without errors",
```
Comment (ppapapetrou76) [inline pkg/deployment/kibana/resync_test.go]:
```suggestion
			name: "Succeeds to re-synchronize all Kibana instances without errors",
```

File: cmd/deployment/kibana/resync.go [hunk 4]
```diff
@@ -28,14 +28,31 @@ import (
 )
 
 var resyncKibanaCmd = &cobra.Command{
-	Use:     "resync <cluster id>",
+	Use:     "resync [<deployment id> | --all]",
```
Comment (marclop) [inline cmd/deployment/kibana/resync.go]: As per https://github.com/elastic/ecctl/issues/44#issuecomment-562053575
```suggestion
	Use:     "resync {<deployment id> | --all}",
```

Comment (karencfv) [reply cmd/deployment/kibana/resync.go]: ah nice! I had no idea 😄

**Pull request merged**